### ISSUE TITLE
installer_mac/make_installer.sh bakes an installer that doesn't remove previous .component .vst .vst3

### ISSUE THREAD

**Issue by esaruoho** [kurasu/surge#239]
Steps to replicate:
1. .`/build-osx.sh`
2. `cd installer_mac`
3. `./make_installer.sh todaysbuild`
4. `cd installer`
5. run the installer-package, be prompted for a password in the right place, finish installer.

expected result:
`/Library/Audio/Plug-Ins/Component` & `VST` & `VST3` now have the newest build.

current result:
if these folders had a previous Surge.component, Surge.vst or Surge.vst3 - they don't get overwritten.

@baconpaul @kzantow  overall, what am i doing wrong? Most users will be getting the installer, install it, and go "okay, i've installed it" == they're gonna be a bit disappointed.

**Comment by kzantow** [comment]
Seems bad but I'm not sure about the installer apps -- @asimilon do you know how the Mac installer is supposed to work to ensure stuff is overridden?

**Comment by baconpaul** [comment]
Also no idea. But agree it seems like the wrong behavior.

But seems, from man pkgbuild, it will be related to this in the info.plist

BundleOverwriteAction specifies how an existing version of the bundle on
     disk should be handled when the version in the package is installed. If
     you specify upgrade, the bundle in the package atomically replaces any
     version on disk; this has the effect of deleting old paths that no longer
     exist in the new version of the bundle.  If you specify update, the bun-
     dle in the package overwrites the version on disk, and any files not con-
     tained in the package will be left intact; this is appropriate when you
     are delivering an update-only package.  Another effect of update is that
     the package bundle will not be installed at all if there is not already a
     version on disk; this allows a package to deliver an update for an app
     that the user might have deleted.

that probably needs to be set to "upgrade" but that would require a test.

**Comment by baconpaul** [comment]
Yeah that's probably it; the pkgbuild --analyze makes an empty info.plist. We need to modify it to include the bundleoverwriteaction

can't do that now. but I bet that's the answer

**Comment by baconpaul** [comment]
Probably need to modify the installer script to do something like this

```
paul:~/dev/music/surge/installer_mac$ cat s.plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array/>
</plist>
paul:~/dev/music/surge/installer_mac$ /usr/libexec/PlistBuddy -c "Add :Parent:BundleOverwriteAction string" s.plist 
paul:~/dev/music/surge/installer_mac$ /usr/libexec/PlistBuddy -c "Set :0:BundleOverwriteAction upgrade" s.plist 
paul:~/dev/music/surge/installer_mac$ cat s.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
	<dict>
		<key>BundleOverwriteAction</key>
		<string>upgrade</string>
	</dict>
</array>
</plist>
```

**Comment by baconpaul** [comment]
http://thegreyblog.blogspot.com/2014/06/os-x-creating-packages-from-command_2.html

and then for a few other of those

**Comment by baconpaul** [comment]
Ahh the problem is that the script is using --analyze incorrectly. You need to point analyze at the parent of surge, not surge

**Comment by baconpaul** [comment]
OK figured it out and have a fix. Installs now overwrite. PR coming which I really need other people to review and test.
